src/main/java/org/opensearch/security/transport/SecurityRequestHandler.java

```diff
@@ -130,7 +130,7 @@ protected void messageReceivedDecorate(
 
             String channelType = transportChannel.getChannelType();
 
-            if (!channelType.equals("direct") && !channelType.equals("transport")) {
+            if (!channelType.equals("direct") && !channelType.equals("transport") && !channelType.equals("stream-transport")) {
```

Comment:
Let's create a set of these values and then check that the channelType is not within the set.